src/credentials/sso/diskCache.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs'
+import * as crypto from 'crypto'
+import { SsoCache } from './ssoCache'
+import { ClientRegistration } from './clientRegistration'
+import { join } from 'path'
+import { homedir } from 'os'
+import { AccessToken } from './accessToken'
+
+export class DiskCache implements SsoCache {
+    private cacheDir: string = join(homedir(), '.aws', 'sso', 'cache')
+
+    loadClientRegistration(ssoRegion: string): ClientRegistration | null {
+        if (!this.registrationExists(ssoRegion)) {
+            return null
+        }
+        const registration = JSON.parse(
+            fs.readFileSync(join(this.cacheDir, `aws-toolkit-vscode-client-id-${ssoRegion}.json`)).toString()
+        )
+        if (registration && this.isNotExpired(registration)) {
+            return registration
+        } else {
+            return null
+        }
+    }
+
+    saveClientRegistration(ssoRegion: string, registration: ClientRegistration) {
+        fs.mkdirSync(join(homedir(), '.aws', 'sso'), { recursive: true })
+        fs.mkdirSync(this.cacheDir, { recursive: true })
+        fs.writeFileSync(
+            join(this.cacheDir, `aws-toolkit-vscode-client-id-${ssoRegion}.json`),
+            JSON.stringify(registration),
+            { mode: 0o600 }
+        )
+    }
+    invalidateClientRegistration(ssoRegion: string) {
+        if (this.registrationExists(ssoRegion)) {
+            fs.unlinkSync(join(this.cacheDir, `aws-toolkit-vscode-client-id-${ssoRegion}.json`))
+        }
+    }
+    loadAccessToken(ssoUrl: string) {
+        if (this.tokenExists(ssoUrl)) {
+            return JSON.parse(fs.readFileSync(join(this.cacheDir, this.accessTokenCache(ssoUrl))).toString())
+        }
+    }
+
+    saveAccessToken(ssoUrl: string, accessToken: AccessToken) {
+        const fileName = this.accessTokenCache(ssoUrl)
+        fs.writeFileSync(join(this.cacheDir, `${fileName}.json`), JSON.stringify(accessToken), { mode: 0o600 })
+    }
+
+    invalidateAccessToken(ssoUrl: string) {
+        if (this.tokenExists(ssoUrl)) {
+            fs.unlinkSync(join(this.cacheDir, this.accessTokenCache(ssoUrl)))
```

Comment:
If accessTokenCache returned the full path, the path join logic wouldn't need to be duplicated.

Reply:
Good catch, much cleaner now